code/modules/mob/living/simple_animal/hostile/terror_spiders/queen.dm

```diff
@@ -210,14 +210,12 @@
 							neststep = 4
 						else
 							spider_lastspawn = world.time
-							var/num_purple = CountSpidersDetailed(FALSE, list(/mob/living/simple_animal/hostile/poison/terror_spider/purple))
-							var/num_white = CountSpidersDetailed(FALSE, list(/mob/living/simple_animal/hostile/poison/terror_spider/white))
-							var/num_brown = CountSpidersDetailed(FALSE, list(/mob/living/simple_animal/hostile/poison/terror_spider/brown))
-							if(num_purple < 4)
+							var/spider_array = CountSpidersDetailed(FALSE)
```

Comment:
`var/list/spider_array` here. It's a list that is being returned so best to type it as well